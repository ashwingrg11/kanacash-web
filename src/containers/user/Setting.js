import React, { Component } from "react";
import { Row, Col, Card, Accordion, Form } from "react-bootstrap";
import { withRouter } from "react-router-dom";
import { Formik } from "formik";
import * as yup from "yup";
import * as api from "../../services/axios/authApi";
import ButtonSpinner from "../../components/ButtonSpinner/ButtonSpinner";
import PropTypes from "prop-types";

class Setting extends Component {
  changePassword = (value, { resetForm, setSubmitting }) => {
    setSubmitting(true);
    api
      .changePassword(value)
      .then((res) => {
        setSubmitting(false);
        if (res.status === 200) {
          this.props.handleChange({ success: true, data: res.data });
          resetForm();
        } else {
          this.props.handleChange({ success: false, data: res.data });
        }
      })
      .catch((err) => {
        console.log("res", err);

        setSubmitting(false);
        console.log("error in changeing password", err);
      });
  };
  render() {
    const lowercaseRegex = /(?=.*[a-z])/;
    const uppercaseRegex = /(?=.*[A-Z])/;
    const numericRegex = /(?=.*[0-9])/;
    const charactersRegex = /(?=.*[\^$*.[\]{}()?“!@#%&/><’:;|_~`])/;
    const schema = yup.object({
      oldPassword: yup
        .string()
        .required("This field cannot be empty.")
        .matches(
          lowercaseRegex,
          "Password must contain one lowercase character."
        )
        .matches(
          uppercaseRegex,
          "Password must contain one uppercase character."
        )
        .matches(numericRegex, "Password must contain one number.")
        .matches(
          charactersRegex,
          "Password must contain one special character."
        )
        .min(8, "Password must be at least 8 characters long."),
      newPassword: yup
        .string()
        .required("This field cannot be empty.")
        .matches(
          lowercaseRegex,
          "Password must contain one lowercase character."
        )
        .matches(
          uppercaseRegex,
          "Password must contain one uppercase character."
        )
        .matches(numericRegex, "Password must contain one number.")
        .matches(
          charactersRegex,
          "Password must contain one special character."
        )
        .min(8, "Password must be at least 8 characters long."),
      repeatPassword: yup
        .string()
        .required("This field cannot be empty.")
        .oneOf([yup.ref("newPassword")], "Password does not match."),
    });
    const initialValue = {
      oldPassword: "",
      newPassword: "",
      repeatPassword: "",
    };
    return (
      <>
        <Row>
          <Col className="activity-title">Settings </Col>
        </Row>
        <Row className="mt-3">
          <Col sm="12" className="card-block">
            <Accordion defaultActiveKey="0">
              <Card>
                <Accordion.Toggle
                  as={Card.Header}
                  eventKey="0"
                  style={{ background: "#fff" }}
                >
                  <h3 className="subhead-title mb-0">Security and Login</h3>{" "}
                </Accordion.Toggle>
                <Accordion.Collapse eventKey="0">
                  <Card.Body>
                    <Row>
                      <Col>
                        <div className="payment-block-wrap theme-form">
                          <Formik
                            validationSchema={schema}
                            onSubmit={this.changePassword}
                            initialValues={initialValue}
                          >
                            {({
                              handleSubmit,
                              handleChange,
                              handleBlur,
                              values,
                              touched,
                              isValid,
                              errors,
                              isSubmitting,
                            }) => (
                              <Form noValidate onSubmit={handleSubmit}>
                                <Form.Group>
                                  <Form.Label>Old Password</Form.Label>
                                  <Form.Control
                                    type="password"
                                    name="oldPassword"
                                    placeholder=" Enter old password"
                                    value={values.oldPassword}
                                    onChange={handleChange}
                                    isValid={
                                      touched.oldPassword && !errors.oldPassword
                                    }
                                    isInvalid={!!errors.oldPassword}
                                    onBlur={handleBlur}
                                  />
                                  {touched.oldPassword &&
                                    errors.oldPassword && (
                                      <Form.Control.Feedback type="invalid">
                                        {errors.oldPassword}
                                      </Form.Control.Feedback>
                                    )}
                                </Form.Group>
                                <Form.Group>
                                  <Form.Label>New Password</Form.Label>
                                  <Form.Control
                                    type="password"
                                    name="newPassword"
                                    placeholder="Enter new password"
                                    value={values.newPassword}
                                    onChange={handleChange}
                                    isValid={
                                      touched.newPassword && !errors.newPassword
                                    }
                                    isInvalid={!!errors.newPassword}
                                    onBlur={handleBlur}
                                  />
                                  {touched.newPassword &&
                                    errors.newPassword && (
                                      <Form.Control.Feedback type="invalid">
                                        {errors.newPassword}
                                      </Form.Control.Feedback>
                                    )}
                                </Form.Group>
                                <Form.Group>
                                  <Form.Label>Confirm Password </Form.Label>
                                  <Form.Control
                                    type="password"
                                    name="repeatPassword"
                                    placeholder="Confirm new password"
                                    value={values.repeatPassword}
                                    onChange={handleChange}
                                    isValid={
                                      touched.repeatPassword &&
                                      !errors.repeatPassword
                                    }
                                    isInvalid={!!errors.repeatPassword}
                                    onBlur={handleBlur}
                                  />
                                  {touched.repeatPassword &&
                                    errors.repeatPassword && (
                                      <Form.Control.Feedback type="invalid">
                                        {errors.repeatPassword}
                                      </Form.Control.Feedback>
                                    )}
                                </Form.Group>

                                <Form.Row>
                                  <ButtonSpinner
                                    isSubmitting={isSubmitting}
                                    title={"Reset Password"}
                                  />
                                </Form.Row>
                              </Form>
                            )}
                          </Formik>
                        </div>
                      </Col>
                    </Row>
                  </Card.Body>
                </Accordion.Collapse>
              </Card>
            </Accordion>{" "}
          </Col>
        </Row>
      </>
    );
  }
}

Setting.propTypes = {
  handleChange: PropTypes.func,
};

export default withRouter(Setting);
