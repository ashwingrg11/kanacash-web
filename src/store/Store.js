import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
