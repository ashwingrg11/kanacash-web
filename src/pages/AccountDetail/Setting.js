import React from "react";
import { Container, Row, Col } from "react-bootstrap";

import MyAccountSidebar from "../../components/MyAccountSidebar";
import { withRouter } from "react-router-dom";
import Header from "../../containers/layout/Header";
import { getBeneficiaries } from "../../store/actions/senderDetail";
import { connect } from "react-redux";
import Footer from "../../containers/layout/Footer";

import AccountHeader from "../../containers/user/AccountHeader";
import Setting from "../../containers/user/Setting";
import CustomAlert from "../../components/Alert/CustomAlert";
import Disclaimer from "../../components/Disclaimer/Disclaimer";
const Settings = () => {
  const [state, setState] = React.useState({
    showAlert: false,
    message: "",
    variant: "",
  });

  const changePassword = (value) => {
    if (value.success) {
      setState({
        showAlert: true,
        message: "Your password has been reset successfully.",
        variant: "success",
      });
    } else {
      setState({
        showAlert: true,
        message: value.data.message,
        // message: "The old password you have entered is incorrect. Please try again.",
        variant: "danger",
      });
    }
  };
  return (
    <React.Fragment>
      <Header />
      <div className="shadow-line"></div>
      <section className="admin-wrapper">
        <Container>
          <Row>
            <Col>
              <AccountHeader />
              <Row className="admin-content">
                <Col xl="3" lg="4" md="5" sm="12">
                  <MyAccountSidebar />
                </Col>
                <Col xl="9" lg="8" md="7" sm="12">
                  <Setting handleChange={(value) => changePassword(value)} />
                </Col>
              </Row>
            </Col>
          </Row>
        </Container>
      </section>
      <Footer />
      {!localStorage.acceptPolicy && <Disclaimer />}

      <CustomAlert
        show={state.showAlert}
        variant={state.variant}
        message={state.message}
        close={() => setState({ showAlert: false })}
      />
    </React.Fragment>
  );
};

const mapStateToProps = (state) => ({
  beneficiary: state.senderDetail.beneficiaries,
  activePage: state.senderDetail.activePage,
  acceptPolicy: state.auth.accept_policy,
});
const mapActionToProps = { getBeneficiaries };

export default connect(mapStateToProps, mapActionToProps)(withRouter(Settings));
