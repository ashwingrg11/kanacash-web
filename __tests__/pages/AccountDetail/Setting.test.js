import React from "react";
import Setting from "../../../src/pages/AccountDetail/Setting";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<Setting />", () => {
  it("Setting renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Setting />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
