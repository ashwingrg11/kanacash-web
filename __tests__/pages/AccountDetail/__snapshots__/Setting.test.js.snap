// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Setting /> Setting renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <div
    className="shadow-line"
  />,
  <section
    className="admin-wrapper"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <h2
            className="account-title"
          >
            My Account
          </h2>
          <p>
            Everything we know about you.
          </p>
          <div
            className="admin-content row"
          >
            <div
              className="col-xl-3 col-lg-4 col-md-5 col-sm-12"
            >
              <ul
                className="sidebar-menu"
              >
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#1077BE",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="activeImg"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Profile
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Beneficiary
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Bank Accounts
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Debit Cards
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Transactions
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Transaction Limit
                  </div>
                </li>
                <li>
                  <div
                    className="d-flex align-items-center"
                    onClick={[Function]}
                    style={
                      Object {
                        "color": "#4F4F4F",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "fontWeight": "bold",
                        "marginBottom": "10px",
                      }
                    }
                  >
                     
                    <img
                      alt="img"
                      src=""
                      style={
                        Object {
                          "marginRight": "10px",
                          "paddingBottom": "3px",
                        }
                      }
                    />
                    Settings
                  </div>
                </li>
              </ul>
            </div>
            <div
              className="col-xl-9 col-lg-8 col-md-7 col-sm-12"
            >
              <div
                className="row"
              >
                <div
                  className="activity-title col"
                >
                  Settings 
                </div>
              </div>
              <div
                className="mt-3 row"
              >
                <div
                  className="card-block col-sm-12"
                >
                  <div
                    className="accordion"
                  >
                    <div
                      className="card"
                    >
                      <div
                        className="card-header"
                        onClick={[Function]}
                        style={
                          Object {
                            "background": "#fff",
                          }
                        }
                      >
                        <h3
                          className="subhead-title mb-0"
                        >
                          Security and Login
                        </h3>
                         
                      </div>
                      <div
                        aria-expanded={null}
                        className="collapse show"
                      >
                        <div
                          className="card-body"
                        >
                          <div
                            className="row"
                          >
                            <div
                              className="col"
                            >
                              <div
                                className="payment-block-wrap theme-form"
                              >
                                <form
                                  className=""
                                  noValidate={true}
                                  onSubmit={[Function]}
                                >
                                  <div
                                    className="form-group"
                                  >
                                    <label
                                      className="form-label"
                                    >
                                      Old Password
                                    </label>
                                    <input
                                      className="form-control"
                                      name="oldPassword"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      placeholder=" Enter old password"
                                      type="password"
                                      value=""
                                    />
                                  </div>
                                  <div
                                    className="form-group"
                                  >
                                    <label
                                      className="form-label"
                                    >
                                      New Password
                                    </label>
                                    <input
                                      className="form-control"
                                      name="newPassword"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      placeholder="Enter new password"
                                      type="password"
                                      value=""
                                    />
                                  </div>
                                  <div
                                    className="form-group"
                                  >
                                    <label
                                      className="form-label"
                                    >
                                      Confirm Password 
                                    </label>
                                    <input
                                      className="form-control"
                                      name="repeatPassword"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      placeholder="Confirm new password"
                                      type="password"
                                      value=""
                                    />
                                  </div>
                                  <div
                                    className="form-row"
                                  >
                                    <button
                                      className="bt btn-secondary btn-secondary-big btn btn-primary btn-block"
                                      disabled={false}
                                      type="submit"
                                    >
                                      Reset Password
                                    </button>
                                  </div>
                                </form>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                   
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
